PORTAL/jobs/_pyperformance.py

```diff
@@ -1114,7 +1282,13 @@ class PyperfTableHeader(_PyperfTableRowBase):
     sources = _PyperfTableRowBase.values
 
     @classmethod
-    def parse(cls, line, filenames=None, *, fail=False):
+    def parse(
+            cls,
+            line: str,
+            filenames: Optional[Iterable[str]] = None,
+            *,
+            fail: bool = False
+    ) -> Optional["_PyperfTableRowBase"]:
```

Comment:
```suggestion
    ) -> Optional["PyperfTableHeader"]:
```

Reply:
This is a classic use case of PEP 673 (the `Self` type), but alas, that's Python 3.11 only.  In the meantime I can fake it by not putting a return type on the base class' `parse` method.

Reply:
You'll also add a comment?